### PR TITLE
Add NewNormalPrecision for constructing a normal distribution based o…

### PR DESCRIPTION
**Pull request by btracey**
…n its precision matrix

### CODE REVIEW

**Comment by btracey** [comment]
Needs https://github.com/gonum/matrix/pull/333

File: distmv/normal.go [hunk 1]
```diff
@@ -77,6 +77,33 @@ func NewNormalChol(mu []float64, chol *mat64.Cholesky, src *rand.Rand) *Normal {
 	return n
 }
 
+// NewNormalPrecision creates a new Normal distribution with the given mean and
+// precision matrix (inverse of the covariance matrix). NewNormalPrecision
+// panics if len(mu) is not equal to prec.Symmetric(). If the precision matrix
+// is not positive-definite, NewNormalPrecision returns {nil,false}.
+func NewNormalPrecision(mu []float64, prec *mat64.SymDense, src *rand.Rand) (*Normal, bool) {
```
Comment (kortschak) [inline distmv/normal.go]: Name the returns.

File: distmv/normal.go [hunk 2]
```diff
@@ -77,6 +77,33 @@ func NewNormalChol(mu []float64, chol *mat64.Cholesky, src *rand.Rand) *Normal {
 	return n
 }
 
+// NewNormalPrecision creates a new Normal distribution with the given mean and
+// precision matrix (inverse of the covariance matrix). NewNormalPrecision
+// panics if len(mu) is not equal to prec.Symmetric(). If the precision matrix
+// is not positive-definite, NewNormalPrecision returns {nil,false}.
```
Comment (kortschak) [inline distmv/normal.go]: Rephrase the return to take into account return parameter names and avoid braces.

**Comment by kortschak** [comment]
Is there a reason to do all this rather than let the client find the inverse (perhaps via a solve)? I'm happy if you think so, then LGTM.

**Comment by btracey** [comment]
There are two reasons:
1) It seems better to have the user say what they want and we do it the right way then to ask for more.
2) It allows more flexibility for changing. If we find this is the wrong way to do it (http://mathoverflow.net/questions/230808/computing-the-inverse-of-a-cholesky-decomposition), clients get the upgraded algorithm for free. Similarly, if we decide storing both is worth it (it may be worth it for machine-learning like applications), clients get that for free too. 

If that seems a good enough reason I'll merge when changes are implemented.

**Comment by kortschak** [comment]
Those arguments are fine. Merge away!

File: distmv/normal.go (hunk 2 above)
Comment (btracey) [inline distmv/normal.go]: Done.

File: distmv/normal.go (hunk 1 above)
Comment (btracey) [inline distmv/normal.go]: Done.

**Comment by kortschak** [comment]
s/ok for false/false for ok/

**Pull request merged**